dipy/align/vector_fields.pyx

```diff
@@ -1475,16 +1479,39 @@ def simplify_warp_function_3d(floating[:, :, :, :] d,
     r"""
     Simplifies a nonlinear warping function combined with an affine transform
 
-    Modifies the given deformation field by incorporating into it a
-    an affine transformation and voxel-to-space transforms associated to
+    Modifies the given deformation field by incorporating into it
+    an affine transformation and voxel-to-space transforms associated with
     the discretization of its domain and codomain.
+
     The resulting transformation may be regarded as operating on the
     image spaces given by the domain and codomain discretization.
     More precisely, the resulting transform is of the form:
 
     (1) T[i] = W * d[U * i] + V * i
 
     Where U = affine_idx_in, V = affine_idx_out, W = affine_disp.
+
+    Parameters
+    ----------
+    d : array, shape (S', R', C', 3)
+        the non-linear part of the transformation (displacement field)
+    affine_idx_in : array, shape (4, 4)
+        the matrix U in eq. (1) above
+    affine_idx_out : array, shape (4, 4)
+        the matrix V in eq. (1) above
+    affine_disp : array, shape (4, 4)
+        the matrix W in eq. (1) above
+    out_shape : array, shape (3,)
+        the number of slices, rows and columns of the sampling grid
+
+    Returns
+    -------
+    out : array, shape = out_shape
+        the deformation field `out` associated with `T` in eq. (1) such that:
+        T[i] = i + out[i]
```

Comment:
This is the actual updated note. The rest is just moving lengthy comments to the `Notes` section.